### PR TITLE
Make S3 bucket script machine runnable

### LINKED ISSUE #40: Improve the automation for the creation of S3 bucket to be machine runnable
### User Story
As a developer, I want to modify the script from issue #7 to be machine runnable, so I don't need an active user to follow the prompts to create the resource.

### Estimated time to complete
4 hours

### Acceptance Criteria
- [x] Update script to be able to run without user input
    - [x] Script should accept a command line argument, initials
    - [x] Script should validate the initials argument, whether it's passed in as a command line argument, or through the user prompt

### PR DESCRIPTION
**Pull request by dansjack**
# Description
Closes #40 

This PR does the following:
* Updates the create S3 bucket script to accept a command line argument, initials, so it can be run by a machine
* Updates the create S3 bucket script to be runnable from any directory
* Updates the create S3 bucket script file extension from `.sh` to `.bash`
* Extracts the initials validation function to a new file, `common.lib`
* Converts `makeReactBucket.json` to `makeReactBucket.yml` so all CloudFormation templates use yml

# Gallery
Successful run of the script using the command line argument
image

Examples of initials validation with the command line argument
image

# Testing
## Pre-reqs
* You need a `bash` or `zsh`. If you're on Windows, there are lots of guides for adding bash, [here's one](https://www.configserverfirewall.com/windows-10/install-bash-on-windows-10/)
* You must have the aws cli installed, and it must be configured to use your access and secret key

## Steps
1. From the root of the repo, run the following: `./devops/makeReactBucket/makeReactBucket.bash <YOUR_INITIALS>` (e.g. `./devops/makeReactBucket/makeReactBucket.bash dj` 
    * initials requirements:
        * must only consist of letters a-z
        * can be no less than 2 chars
        * can be no more than 5 chars
2. You'll see the actual aws command printed in the terminal
3. Once the script finishes, make sure the new bucket has been created in the [S3](https://s3.console.aws.amazon.com/s3/home?region=us-west-2) dashboard.
4. Copy/paste the following into a file called `index.html`:
```html
<!doctype html>

<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Public site</title>
</head>
<body>
  <h1>Hello, world!</hi>
</body>
</html>

```
5. In the Objects tab of your new S3 bucket, click upload and then choose the file you just created, then click upload again
6. Go to the Properties tab of the S3 bucket and scroll to the bottom. Go to the URL for your Bucket website endpoint. If everything worked, you should see the Hello, world! text like in the Gallery above.
7. Once done testing, make sure to empty and delete your S3 Bucket, then delete your CloudFormation stack (in that order)

### CODE REVIEW

**Review by KristinJJ — APPROVED** [review]
Looks good to me! File is visible in S3 bucket after following testing steps to create CloudFormation.

Just curious, is there a reason to change from .sh to .bash script? I saw some comparisons online about how .bash is more flexible, while .sh is easier to learn. Do you think it is worth having all of the scripts in the DevOps folder be .bash or .sh scripts for consistency? (I can change my script to use .bash if this is the case.)

**Comment by dansjack** [comment]
> Just curious, is there a reason to change from .sh to .bash script? I saw some comparisons online about how .bash is more flexible, while .sh is easier to learn. Do you think it is worth having all of the scripts in the DevOps folder be .bash or .sh scripts for consistency? (I can change my script to use .bash if this is the case.)

Hey @KristinJJ, thanks for the review! 

I switched to bash because I installed a shell linting plugin called [ShellCheck](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck), and I got warnings that some of the commands I was using wouldn't work with POSIX. It wouldn't be a bad idea to switch to bash. You shouldn't have to change any of your code, just the file extension.

Also, I downloaded a formatting plugin for shell called [ShellFormat](https://marketplace.visualstudio.com/items?itemName=foxundermoon.shell-format) if interested.

**Review by dale108 — APPROVED** [review]
Looks good

File: devops/makeReactBucket/makeReactBucket.bash [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+# import validateInitials from common.lib
+# shellcheck disable=SC1091
+. "$SCRIPT_PATH/../common.lib"
+
+RANDOM_STRING=$(cat /dev/urandom | base64 | tr -dc 'a-z0-9' | fold -w 5 | head -n 1)
+BUCKET_TEMPLATE="$SCRIPT_PATH/makeReactBucket.yml"
+INITIALS_PROMPT="What are your initials? ('q' to quit)"
+
+initials="$1"
```
Comment (toddysm) [inline devops/makeReactBucket/makeReactBucket.bash:13]: How about having more than one parameter? For example `region` and `initials`?

**Pull request merged**